### ISSUE TITLE
The type checker seems to report only the first error

### ISSUE THREAD

**Issue by luauser32167** [vtereshkov/umka-lang#234]
```
fn main() {
  var x: int32 = "a"; // Error: Incompatible types int32 and str
  var y: int32 = "b"; // not reported
}
```
Reporting more than a single type error per compiler invocation (see this [godbolt/compiler-explorer](https://godbolt.org/#z:OYLghAFBqd5QCxAYwPYBMCmBRdBLAF1QCcAaPECAMzwBtMA7AQwFtMQByARg9KtQYEAysib0QXACx8BBAKoBnTAAUAHpwAMvAFYTStJg1AB9U8lJL6yAngGVG6AMKpaAVxYMQAJlIOAMngMmABy7gBGmMQgAKykAA6oCoS2DM5uHt7xickCAUGhLBFRsZaY1ilCBEzEBGnunj6l5QKV1QR5IeGRMRZVNXUZjX3tgZ2F3dEAlBaorsTI7BwAbqh46ADUBJgKBBAra5PrAKQA7ABCRxoAguu36zvErtbH5%2BuBBOtURwDMZy8AIvcfhdrnc3oJ1qpjt9AQpgZcbnd3usAJ7Q2Hw66nf4caa0TjRXieDhaUioTiOe6zeaYY5eb48UgETS40gQJBoFhxOiRciUTnc%2BhRZDALheHxYJZ4BYANTwmAA7gB5OKMTiMmi0LbEBSUMIs0hhQLVFHq3ictiCJUMWimkm8LAsQzAcT20j4YhlGxLbYGzCqMquLZm8iCTD4t20PBhYgm5xYA0EYh4Fhm6ZUAzABRyxUqtXcXj8QQiMTsKQyQSKFTqN26Lj6Z0gUzGcxRsK6yDTVBxGwCXUcXhNXueCAOAaeev%2BUYFIp6BJJYfjufZYcdGfdetDirDJeb8Ne7dtNddKKbncuep6HZH6cniTTBTUhb3/QEokG8kcSEADgAbABaX9JHWYBkGQdYxQAOi8dYIFwQgSDpBlJl4Zl7WmdkQFmAg4iDPkIAFHliGCVhFlUP9AOA0DwKgrxeEwfAiGTdA9CLYRRHEcs2KrNQDTrUgFVjOI01fDhCVIYlSU/JUg1wj5UCoH8AKAkCwIgrxoNg5wuSIpCuBQpkWUmaYAGsQEkaJIOiLwNDFX9vgZaITm/b8uFiCNJHfN1P0HEANEM9C2RgRAUAwHBGJIfDNUYUiQzYktOOkbilF42tMgMIwm1MdALH3ZoRzHC8MknBh0GPcZiiyBcUl3KqcgYcrZxKPLh1afoionXKrFa4ZGomXo2lq68aj64oHyfRZ9g2LYdggQ5ThBRFbmRKEfkBI5xSYDavExJbwQ%2BNE1rpLwwm23bsQRVkI3EyTeE/WEJqQuiAq0DCkGwuT8MIoUSLYThaHQGjIO%2BDRINJBiEOYzJ4o4ssktkHia1JXQfEEphhILPE3wkj9OBknCg3WBT1gBoGQbBrTUB0oUnoMtDXrZDkqcFXkKAI5miJAUmuGMEH3UwKVZXlZVVWJDU6G1XUIH1N0jWYYg7UZC1GAIa1bQNR1nVdUkPQPH1%2B3BgNkCDRZGXecMDTbWMFfjRZSSTFMRIzJgsxzEX8w1WQErhit5BSpGdHSxtmxytsOzmskexSftBxalJ7FK2qp3yO963neravT1dbwqq845ac90k6rcC5vFPc7PQaOqvXqc9nfSZjmZ8vCxsSvKkzgScBiDgdBjRYPgpjadQozTIkE4rMkLgAE4NHpSQvCn74vGn6QPPbu7OF8/z6dZOBYCZrBB8itnouYP6C1972JHhyt/b4kBvgbTK4iYBQOK6g87FHRPq5Ksq64bmkFnGqf86rZ3LvXaQJcGBtVqGAmBcDRo3wGu1IuNcy5jCgeNJuixXAMEIETWSQZdrm2IFQJgCwEQpjiPQS0VRhwIgRHEYgqAFjoDmLSGauxJi7SWNUBEYJ1D7UwMASIu0wSmhEWI4gu0IjAECIIu4UIQBHTABwJgTB1ESLuGiVRMJ1jqLCGEbRvwEQOF2g4SCl1W43VxhwZQb8OJUlwUhb4I9AqYQ%2BnhNm31Ii/TIhRFSVA4jgW%2BNBaC9EIpQyfjDUsKDkrVgfk/NGGMeC2I3mSPGxD5KKXIspYCISwkRJghAbSLNiBuLpkZRmoVqas35BzIUIAinLwlALaUmA3Z5jFoWCWkQpYy1JHLE0IZlZWhtHaHWmAnRGG1g6PAnpyj6z9EbE2IZzYRlJFbOMYVEzJlTJjPgmZszCx6XFL2sMEkI3vmlJ%2BGVQCv3fmIfQ0Zw5dijn2Len98oJycGA5OWCNzgNAegtOK4UjIL3N1Q8aDLzQq/rA2ukDgXDXgWC1BIwUWnhwTSR%2BGScbeU7vkyinxQnrHCRpUpx9KkbQctU9CY8Z5g2nq5b809ojfgcn%2Bb80R3KcE8oSjuA4LB%2BRenvYKmE0BH2iVFHk59TaFkufEriNyklpTXo2AAXngYAPzhx/KTqVZB34QUCEzhC3IgCoimsQYXeF%2BrYVYqBTazFQ1kUupAN%2BXFz4OAAHo/XrAACoolVKiak6xpW0gQJETApAiaVNoKgJgGxDDrH9KwWhmBrHXH9AkGonwGCbG2AQYwc11hTReItME/DKnCLwMvdEx0trih0bcWtqIQBvEbUdbap1W1mNBHcYwTbVBtvWCOo6KJzonH%2BDY0SdiiUcAAFq6pcTSJCgrd5vSwjkr6TT/GxQ4Dq4A6xQbTzBlEyGaxzJXyuaqu%2B6rka3tSSJa6mTpI5KJopE9Z7IIXv7mUg9tL54MoZlK4D%2B76lRC1RoWe/NBZdLOaLOK/SdR6gNCMhWYyqb0LVlMh0MytZ2wWUs70vo3T%2BkDMGS%2BmzLbRmtiiW2%2BzHZHOdq7ZDHslXFnvbfP2T7A6asyie157Z4AfOHDHR139CoYsBeuU8ZrUhgJAVa7FecYWlzhcVaTSLMEKYwdp4uHqDMN0fLg8yBLbpZK/L%2B89FM4LRM3WB3ETKJ5cuns5LgJwF6/i8FwDQTlRKCusz5UVO9R6iWeqF7527pg%2Bh1PHSQQA%3D%3D) page) is good because it saves time.

**Comment by vtereshkov** [comment]
Yes, the compiler (not only the type checker) always stops on the first error and does not try to recover.

**Comment by marekmaskarinec** [comment]
I'm personally like it this way. A lot of compilers that report multiple errors can report errors that only exist because of other errors.
